aten/src/ATen/mps/EmptyTensor.cpp

```diff
@@ -65,13 +76,25 @@ TensorBase empty_strided_mps(
     IntArrayRef stride,
     ScalarType dtype,
     c10::optional<Device> device_opt) {
-  auto device = device_or_default(device_opt);
-  TORCH_INTERNAL_ASSERT(device.is_mps());
-  const DeviceGuard device_guard(device);
-  auto* allocator = at::mps::GetMPSAllocator();
-  constexpr c10::DispatchKeySet mps_dks(c10::DispatchKey::MPS);
-  return at::detail::empty_strided_generic(
-      size, stride, allocator, mps_dks, dtype);
+#if defined(__APPLE__)
+#if __is_target_os(macOS)
+  if (__builtin_available(macOSApplicationExtension 12.3, *)) {
+    auto device = device_or_default(device_opt);
+    TORCH_INTERNAL_ASSERT(device.is_mps());
+    const DeviceGuard device_guard(device);
+    auto* allocator = at::mps::GetMPSAllocator();
+    constexpr c10::DispatchKeySet mps_dks(c10::DispatchKey::MPS);
+    return at::detail::empty_strided_generic(
+        size, stride, allocator, mps_dks, dtype);
+  } else {
+    TORCH_INTERNAL_ASSERT(false, "MPS backend supported MacOS 12.3+")
+  }
+#else
+  TORCH_INTERNAL_ASSERT(false, "MPS backend supported MacOS 12.3+")
```

Comment:
Can we make the two below clear that these are that pytorch was built without MPS support.

Not an internal assert either.

Reply:
Okay.